torch/testing/_internal/distributed/nn/api/remote_module_test.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python3
+import enum
+import uuid
+from typing import Tuple
+
+import torch
+import torch.testing._internal.dist_utils as dist_utils
+from torch import Tensor, nn
+from torch._jit_internal import Future
+from torch.distributed.nn.api.remote_module import (
+    RemoteModule,
+    _gen_global_unique_name,
+    _RemoteModule,
+)
+from torch.testing._internal.distributed.rpc.rpc_agent_test_fixture import (
```

Comment:
Can we test using invalid module_creators throw an appropriate exception? Examples:

1. module_creator is a class that is not type of nn.Module.
2. module_creator returns an object that is not type of nn.Module.

Reply:
Adding corresponding tests.